draft-pauly-httpbis-geoip-hint.md

```diff
@@ -73,63 +101,102 @@ For example, the header for an entry "192.0.2.5,US,US-AL,Alabaster" would be:
+When the client is routing traffic through a proxy or a VPN, the
+IP address used to generate this geolocation hint MUST be an
+address that is presented upstream beyond the proxy or VPN
+(in other words, the "egress IP address"). The proxy or VPN's
+selection of this egress IP address MAY have been based on
+the client's original un-proxied IP address, but any hints that
+the client presents to servers beyond a proxy or VPN MUST NOT
+reveal more geolocation information that would be possible to
+determine from looking up information about the egress IP address
+itself.
+
+The client MAY include the client hint header in requests to the
+server after the server has explicitly opted in to receiving the
+hint, or if the client knows of specific server configurations,
+such as proxy settings, that support including the hint.
 
-- Use the client hint information instead of consulting another geolocation feed.
-- Check the value of the "feed" parameter on the header and determine if it is a trusted feed.
-   If this feed is trusted, but is not the default feed used by the server, the server
-   can choose to prefer the feed indicated by the client.
-- Check the value of the "feed" parameter on the header and fetch a copy of the feed
-   to verify the mapping, if a copy of the feed has not been fetched recently.
-- If the feed indicated in the "feed" parameter is unknown or untrusted, but starts
-   becoming common, the server can flag this feed as one to be manually checked and
-   added, if appropriate. This allows servers to automatically discover when new
-   feeds and services are brought up..
-
-If the server is acting as a forward proxy, such as a CONNECT proxy, it can use the hint
-to determine an appropriate geo-mapped IP address to use for outbound connections, or a
-client subnet to present in the EDNS0 Client Subnet extension for DNS queries {{?RFC6891}}
-{{?RFC7871}}.
-
-# Security Considerations {#security}
-
-The use of the IP Geolocation Client Hint MUST use the Sec- header prefix as recommended
-in {{!RFC8942}}.
-
-Servers MUST NOT use IP Geolocation Client Hints for making security or access-control decisions,
-as the value can be spoofed by a client. The hint is intended only for use in optimizing behavior.
+# Server Behavior
 
-The value contained in this hint SHOULD be based only on a IP Geolocation feed value for
-an IP address the client is already presenting to a server. In order to avoid disclosing
-any private information, this value MUST not be based on geolocation of the client determined
-by other means, such as physical latitude and longitude coordinates.
+Upon receiving a Geolocation Client Hint, a server can use the
+information to influence its behavior in various ways, such as
+determining the content of HTTP responses.
+
+Servers can choose to use the hint value in one of several ways,
+including:
+
+- Using the client hint information instead of consulting IP-based
+geolocation feeds.
```

Comment:
```suggestion
geolocation feeds.
- Recognizing a mismatch between the client hint information and the server's
current result from its IP-based geolocation feed as a reason to schedule an
automatic refresh of its geolocation feed information. This can help ensure that
changes to feeds are adopted quickly, improving results for clients that don't
send the client hint.
```